src/mrinufft/operators/interfaces/gpunufft.py

```diff
@@ -502,6 +502,19 @@ def uses_sense(self):
         """Return True if the Fourier Operator uses the SENSE method."""
         return self.raw_op.uses_sense
 
+    def update_smaps(self,new_smaps):
+        """Update pinned smaps from new_smaps.
+
+        Parameters
+        ----------
+        new_smaps: np.ndarray
+            the new sensitivity maps
+
+        """
+        smaps_ = new_smaps.T.reshape(-1, new_smaps.shape[0])
+        np.copyto(self.raw_op.pinned_smaps, smaps_)
+        self.smaps = new_smaps
+
```

Comment:
I would rather recommend to have this as a property. Have a look at how samples.setter is dont just below. What we need is that the update of Smaps triggers all these compute internally. We dont want the user to do call this function.

Reply:
Make sure to have the property defined in base class, with a setter function raising an error (just in case, if we have an interface which does not support smaps update). This setter can be overridden in respective downstream inherited classes later on.